src/main/java/uk/co/onsdigital/discovery/dao/DatasetDAOImpl.java

```diff
@@ -105,28 +126,29 @@ public void createOrUpdateMetadata(DatasetMetadata form) throws MetadataEditorEx
         try {
             int minorVersion = isEmpty(form.getMinorVersion()) ? 0 : Integer.parseInt(form.getMinorVersion());
 
-            SqlParameterSource sqlParameterSource = createParameterSource.apply(toArray(
-                    namedParam(JSON_METADATA_FIELD, form.getJsonMetadata()),
-                    namedParam(MAJOR_VERSION_FIELD, Integer.parseInt(form.getMajorVersion())),
-                    namedParam(MINOR_VERSION_FIELD, minorVersion),
-                    namedParam(REVISION_NOTES_FIELD, form.getRevisionNotes()),
-                    namedParam(REVISION_REASON_FIELD, form.getRevisionReason()),
-                    namedParam(DATASET_ID_FIELD, UUID.fromString(form.getDatasetId()))
-            ));
+            SqlParameterSource sqlParameterSource = createParameterSource.apply(
+                    new NamedParam.ListBuilder()
+                            .addParam(JSON_METADATA_FIELD, form.getJsonMetadata())
+                            .addParam(MAJOR_VERSION_FIELD, Integer.parseInt(form.getMajorVersion()))
+                            .addParam(MINOR_VERSION_FIELD, minorVersion)
+                            .addParam(REVISION_NOTES_FIELD, form.getRevisionNotes())
+                            .addParam(REVISION_REASON_FIELD, form.getRevisionReason())
+                            .addParam(DATASET_ID_FIELD, UUID.fromString(form.getDatasetId()))
+                            .toList()
+            );
 
             namedParameterJdbcTemplate.update(UPDATE_METADATA_QUERY, sqlParameterSource);
-        } catch (DataAccessException ex) {
-            throw new MetadataEditorException(DATABASE_ERROR, ex);
+        } catch (Exception ex) {
+            MetadataEditorException.ErrorCode errorCode = UNEXPECTED_ERROR;
+
+            if (ex instanceof DataAccessException) {
+                errorCode = DATABASE_ERROR;
+            } else if (ex instanceof NumberFormatException) {
+                errorCode = STRING_TO_INT_ERROR;
+            }
+            throw new MetadataEditorException(errorCode, ex);
```

Comment:
I considered that but its doing the same thing. The only thing thats different is the errorcode. Seemed redundant to repeat the same code just for that.